streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -293,6 +328,8 @@
      *                          if not specified the default serdes defined in the configs will be used
      * @param <V1>              the value type of the other stream
      * @param <R>               the value type of the new stream
+     *
+     * @return a {@link KStream} containing the join-tuples computed by the given {@link ValueJoiner}
```

Comment:
I did not have a better idea than "join-tuple" -- I agree that it is not the best term...

However, the result is still key value pairs: `(k1,v1) join (k1,v2) => (k1, v')` with `v' = valuejoin(v1,v2))`.